### ISSUE TITLE
Spaltlampenbefund 

### ISSUE THREAD

**Issue by Toelk-A** [chrbusch/GTM-V#424]

**Comment by EmmaSchroederGBMV** [comment]
Hallo Herr Dr. Busch,

ich habe noch ein paar Dinge zu den Spaltlampenbefunden. 

Und zwar erscheint nicht immer aber ab und zu wenn man auf Bearbeiten klickt, dass kein passender Datensatz gefunden wurde…allerdings müsste dieser vorhanden sein. Bsp 2018-0601-CD
![sl-befund_kein_passender_datensatz_gefunden](https://user-images.githubusercontent.com/43231378/45817157-225bc000-bcde-11e8-8d61-f9e232508312.png)

Zudem erscheinen in der Liste der SL-Befunde auch Befunde die nicht ausgefüllt wurden da die HH beispielsweise Verworfen oder in QS sind, das passiert weil man den Befund der HH fälschlicherweise geöffnet hat. Trotzdem darf das nicht in der Liste für die Unterschriften auftauchen. 
![sl-liste_mit_leeren_befunden](https://user-images.githubusercontent.com/43231378/45817208-4d461400-bcde-11e8-85eb-500771546977.png)

Dann werden leider die Graphiken übernommen…und zwar wenn die HH-CS zentrale Narben hat, die HH CD aber nicht, erscheint das trotzdem in der Graphik für CD sobald es bei CS eingegeben wurde, oder andersherum. Sie sind also irgendwie miteinander verknüpft.
![sl-befund_mit_narbe_und_richtiger_graphik](https://user-images.githubusercontent.com/43231378/45817379-e4ab6700-bcde-11e8-931a-0be7d5c64737.png)
![sl-befund_ohne_narbe_mit_falscher_graphik](https://user-images.githubusercontent.com/43231378/45817387-ed9c3880-bcde-11e8-99c2-61adbb68658d.png)

Wenn über die HH CD ein Spaltlampenbefund erstellt wird und danach auf Partnerhornhaut geklickt wird um Befund der HH CS zu erstellen, ist in den Prozessschritten der HH CS kein SL-Befund hinterlegt.
![prozessschritte_mit_sl-befund_eingabe_richtig](https://user-images.githubusercontent.com/43231378/45817511-52579300-bcdf-11e8-9404-7f5b68fc682d.png)
![prozessschritte_mit_sl-befund_mit_falschem_tag_der_eingabe](https://user-images.githubusercontent.com/43231378/45817526-597ea100-bcdf-11e8-93ac-67d73e2b7961.png)
Das rot markierte war ich erst heute….

Zudem war die Frage ob wir in der Liste für die Spaltlampenbefunde noch die Befundarten, Eingangs-/ Ausgangsbefund abbilden können.

**Comment by chrbusch** [comment]
Danke für den langen Bericht! Dringende Bitte: NUMMERIEREN Sie Ihre Punkte, sonst ist es recht mühsam, zu antworten.
Frage: "Und zwar erscheint nicht immer aber ab und zu wenn man auf Bearbeiten klickt,.."
Wo genau klicken Sie auf "Bearbeiten"?

**Comment by chrbusch** [comment]
"...dass kein passender Datensatz gefunden wurde…allerdings müsste dieser vorhanden sein. Bsp 2018-0601-CD"
Um dies zu prüfen bräuchte ich bitte eine HH, für die es nach wie vor keinen Befund gibt, die erwähnte HH hat aber nun einen.

**Comment by chrbusch** [comment]
"..erscheinen in der Liste der SL-Befunde auch Befunde die nicht ausgefüllt wurden"

Ich zeige jetzt nur noch Befunde, bei denen mind. eines der ersten drei Felder ausgefüllt ist.

**Comment by chrbusch** [comment]
"Dann werden leider die Graphiken übernommen.."

Ist meines Erachtens behoben, bitte KRITISCH testen - wichtig!

**Comment by EmmaSchroederGBMV** [comment]
1. Frage: "Und zwar erscheint nicht immer aber ab und zu wenn man auf Bearbeiten klickt,.."
Wo genau klicken Sie auf "Bearbeiten"?
Wenn man in den Prozessschritten den Spaltlampenbefund öffnet und dann Bearbeiten klickt.
![sl-befund_feld_bearbeiten](https://user-images.githubusercontent.com/43231378/45871046-b4ba9d00-bd8c-11e8-92b2-d05823627412.png)

**Comment by EmmaSchroederGBMV** [comment]
2. "...dass kein passender Datensatz gefunden wurde…allerdings müsste dieser vorhanden sein. Bsp 2018-0601-CD"
Um dies zu prüfen bräuchte ich bitte eine HH, für die es nach wie vor keinen Befund gibt, die erwähnte HH hat aber nun einen.

Dazu können Sie sich die 2018-0593-CS anschauen, dort ist das Problem noch eindeutig erkennbar.
Am 17.09. wurden über HH CD beide SL-Befunde erstellt, dies ist aber nicht in den Prozessschritten der HH CS sichtbar.

**Comment by EmmaSchroederGBMV** [comment]
3. "..erscheinen in der Liste der SL-Befunde auch Befunde die nicht ausgefüllt wurden"
Ich zeige jetzt nur noch Befunde, bei denen mind. eines der ersten drei Felder ausgefüllt ist.
Und vorne die Befundart

Das habe ich getestet und klappt, auch in der Liste wird die Befundart abgebildet.

**Comment by EmmaSchroederGBMV** [comment]
4. "..ist in den Prozessschritten der Partner-HH CS kein SL-Befund hinterlegt."
Erledigt, bitte testen

Das habe ich ebenfalls getestet und funktioniert nun.

**Comment by EmmaSchroederGBMV** [comment]
5. "Dann werden leider die Graphiken übernommen.."
Ist meines Erachtens behoben, bitte KRITISCH testen - wichtig!

Klappt nur halb, ist bei CD die Graphik nicht erforderlich, werden nach erneutem Öffnen die bei CS eingetragenen Punkte erst wieder angezeigt, nach dem das Feld "Graphik starten" nochmals angeklickt wurde.

**Comment by chrbusch** [comment]
Programmierfehler mit "kein Datensatz" gefunden und behoben!

**Comment by EmmaSchroederGBMV** [comment]
Ja, jetzt passt es. Am Freitag gibt es eine erneute Testung während der Spaltlampenuntersuchung.

**Comment by EmmaSchroederGBMV** [comment]
Die Testung war auch erfolgreich, das läuft nun ohne weitere Probleme. 

Die Frage ist aber ob die Daten bezüglich Pseudophakie, Arcus senilis und opt. Zone automatisch mit übernommen werden können.

**Comment by chrbusch** [comment]
Pseudophakie, Arcus senilis und opt. Zone
werden nun autom. in den SpLa-Befund übernommen

**Comment by EmmaSchroederGBMV** [comment]
Ich habe mir das angeschaut, leider wird der Arcus nicht übernommen, der Rest schon.
![sl-befund_kein a senilis](https://user-images.githubusercontent.com/43231378/47487939-ba9c1600-d843-11e8-9982-1d0daf047f9d.png)

**Comment by chrbusch** [comment]
Das ist korrigiert, aber folgendes sollte m.E. angepasst werden:
Bei den Transplantatdetails hat das Feld Arcus nur "ja/nein"; beim SpLaBefund gibt es hingegen "Nein
Ja, zart
Ja, ausgeprägt"
Soll ich diese feinere Liste auch bei den Transplantatdetails  anzeigen?
